Globalize and rename has_locked_chest_privilege

Thomas--S opened the pull request:
Why?

- Many mods have to use a similiar function
- It is annoying to copy&paste all changes made to this function made in mtg to other mods
- It is a first step to customize the key system. I will maybe build on this base here later.

sofar commented:
- not opposed to the idea of exporting this function
- suggest something less unwieldy like `can_interact_with_node()`
- since this particular interface is optimized for internal use of `meta` I would break that and instead pass `pos` instead of `meta` so it can be used much more generically. So you would make it `default.can_interact_with_node(pos, player)` (or swap player and pos)
- if you look at the code, this isn't generally so bad since almost all calls to this function fetch `meta` right before this call, so it would actually clean up a bit more.

my biggest concern is that this effectively adds a significant API to the existing protection API. This shouldn't happen without a larger discussion about how to move that forward.

Thomas--S commented:
I think I should have addressed all your suggestions now. Thanks for helping!

I added the comment "No API" because for now, this PR is mainly thought as a base for further generalization of the interact protection system. I think that the `<quote>`_larger discussion_`</quote>` should happen before declaring it an official API, but after the further planned changes.

sofar commented:
Yes, see, that kills like all but the one get_meta() calls :)

So far so good. We should seriously consider this.

Thomas--S commented:
@paramat, @rubenwardy, @sofar: Could you please review this? I need this PR merged to be able to work on the further improvements. Thanks in advance.

rubenwardy commented:
Maybe move `default.can_interact_with_node(player, pos)` to functions.lua?

Thomas--S commented:
@rubenwardy I plan to do this, when I propose this function to be added to the official API. For now, this is only meant as a base for my further improvement suggestions. This here should be as lightweight as possible, IMHO.

paramat commented:
Moving it to functions.lua won't make it less lightweight, nodes.lua really isn't the place for this.
I'd prefer it in functions.lua. All else looks fine to me.

Thomas--S commented:
PR updated according to @rubenwardy's and @paramat's suggestions.

paramat commented:
:+1:  as long as it was tested either before or after this latest update.

Thomas--S commented:
@paramat I tested the latest version, and it should work. However, I did not test every edge-case.

paramat commented:
No problem +1.

sofar commented:
So I'm :+1: on the code and this patch.

I do think we should not declare it an outside API. Perhaps leave a note as such, and leave it undocumented in game_api.txt.

Thomas--S commented:
Thank you very much for merging!
